apps/s_time.c

```diff
@@ -240,16 +224,10 @@ int s_time_main(int argc, char **argv)
                                    www_path);
             if (buf_len <= 0 || SSL_write(scon, buf, buf_len) <= 0)
                 goto end;
-            while ((i = SSL_read(scon, buf, sizeof(buf))) > 0 ||
-                        SSL_get_error(scon, i) == SSL_ERROR_WANT_READ ||
-                        SSL_get_error(scon, i) == SSL_ERROR_WANT_WRITE)
```

Comment:
There are two problems here. The first is a bug in libssl (fix is in #3954). The second is slightly less obvious. Once the libssl bug is fixed this should make s_time with "-reuse" appear to work - but it won't actually be doing resumption at all! The reason is that it is not waiting for the NewSessionTicket to arrive before trying to use the session. See the "Sessions" section of my blog post for details: https://www.openssl.org/blog/blog/2017/05/04/tlsv1.3/

This is actually quite problematic. If we are talking to an OpenSSL server then we can be confident that the NewSessionTicket will arrive quite quickly. However other servers may not do so (or in theory they may not send one at all). For example, IIRC picotls doesn't send the NewSessionTicket until the server sends some application data - which might be some time (or not at all). At least that's the way it used to work - I've not tried testing against it recently.